src/site/content/en/blog/codelab-building-a-sidenav-component/index.md

```diff
@@ -0,0 +1,332 @@
+---
+layout: codelab
+title: "Codelab: Building a Sidenav component"
+authors:
+  - adamargyle
+description: |
+  Learn how to build a responsive slide out side navigation layout component.
+date: 2020-11-25
+hero: hero.jpg
+thumbnail: thumb.jpg
+glitch: gui-challenges-sidenav-lab
+glitch_path: app/index.html
+related_post: building-a-sidenav-component
+tags:
+  - css
+  - dom
+  - javascript
+  - layout
+  - mobile
+  - ux
+---
+
+This codelab teaches you how to build a responsive slide out side navigation layout component
+on the web. We'll build the component as we go, starting with HTML, then CSS,
+then JavaScript.
+
+Check out my blog post [Building a Sidenav component](/building-a-sidenav-component)
+to learn about the CSS tools chosen for building this component.
+
+{% YouTube 'PzvdREGR0Xw' %}
+
+## Setup
+
+{% Instruction 'remix', 'ol' %}
+1. Open `app/index.html`.
+
+## HTML
+
+First, get the essentials of the HTML setup so there's content and some boxes to work with.
+I'll be using an `<aside>` to hold the navigation menu, and a `<main>` to hold the primary page content. 
+Drop the following HTML into the body tag.
+
+```html
+<aside></aside>
+<main></main>
+```
```

Comment:
Remember that codelabs should be action-focused. We want them to learn by doing. We should provide instructions first and foremost, and then context.

```
Drop the following HTML into the body tag.

    …
    <body>
      <aside></aside>
      <main></main>
    </body>
    …

The `<aside>` holds the navigation menu, and the `<main>` holds the primary page content. 
```

* Please review your insructions throughout the codelab and check if you can refactor to this pattern of "instruction first, and then context after the code sample".
* We should provide a little more context in the code samples so that there is no ambiguity about where the reader should put the code. Please review all of your code samples to see if they need a bit more context. (This one is obvious but in general it can quickly become unobvious as you progress through a codelab.)
* Please replace `I` and `I'll` with `you` throughout the post. The focus here is that the reader should do the action. Not us. For this reason we want to make it clear that **you** need to do X.

Reply:
(Meta note: please leave this convo unresolved so I can remember to check back on these points throughout the doc later)